code/game/objects/structures/inflatable.dm

```diff
@@ -350,14 +350,14 @@
 	return TRUE
 
 /obj/structure/inflatable/shelter/update_icon()
-	overlays = list()
-	var/mob/living/carbon/human/occupant = locate(/mob/living/carbon/human) in contents
-	if(occupant)
-		var/image/occupant_overlay = occupant.appearance
-		overlays += occupant_overlay
-	var/image/cover_overlay = image('icons/obj/inflatable.dmi', icon_state = "shelter_top", layer = FLY_LAYER)
-	cover_overlay.plane = ABOVE_HUMAN_PLANE
-	overlays += cover_overlay
+	vis_contents.Cut()
+	overlays.Cut()
+	for(var/mob/living/L in contents)
+		vis_contents.Add(L)
```

Comment:
The thing about this is that you will be rendering a bunch of sprites that you can't see because they're stacked on top of each other. That's not really... great. Jiggle their offsets here if you want to do this.

Reply:
is there a way to with vis_contents that doesn't touch pixel_x or pixel_y? plus there's also some sprites like goliath that nearly take up the whole tile so offsetting them would make them clip outside. this is really just a fluff effect anyways.